### ISSUE TITLE
AppSecException unchecked exception is swallowed by Stream.map() when thrown

### ISSUE THREAD

**Issue by tamasmak** [vaadin/appsec-kit#124]
AppSecException is an unchecked exception. When thrown inside the [`OpenSourceVulnerabilityClient::queryBatch()`](https://github.com/vaadin/appsec-kit/blob/main/appsec-kit-backend/src/main/java/com/vaadin/appsec/backend/OpenSourceVulnerabilityClient.java#L64) method call during [`OpenSourceVulnerabilityService::getVulnerabilities()`](https://github.com/vaadin/appsec-kit/blob/main/appsec-kit-backend/src/main/java/com/vaadin/appsec/backend/OpenSourceVulnerabilityService.java#L59) method call it is swallowed by the `Stream::map()`.

Affected AppSec Kit versions: 1.0.0, 2.0.0, 3.0.0

**Comment by sosa-vaadin** [comment]
The exception is not swallowed by `Stream.map()`. The mapping operation is interrupted and the exception is thrown.

If you replace the line of code in `VulnerabilityStore.java:77` with:
```java
        List<OpenSourceVulnerability> vulnerabilities = null;
        try {
            vulnerabilities = osvService
                    .getVulnerabilities(components);
        } catch (Exception e) {
            LOGGER.error("There was an error scanning vulnerabilities", e);
        }
```

You can see the exception is caught and the error message is logged.

What swallows the exception is the click listener of the button that started the scan. What we need to discuss is where to add the exception handling code and what to do when an exception is caught.

Another thing to note is that the most common exception thrown here is the read timeout exception addressed in #30.

**Comment by sosa-vaadin** [comment]
Since `AppSecService#scanForBulnerabilities()` returns a `CompletableFuture`, we can leverage that to let the user handle a failed completion. An example is done in the `AppSecView` "scan now" button.

**Comment by tamasmak** [comment]
Reopening the ticket for backporting the fix to 1.0 and 2.0 versions too.

**Comment by tamasmak** [comment]
Cherry-picked to [1.0](https://github.com/vaadin/appsec-kit/pull/151) and [2.0](https://github.com/vaadin/appsec-kit/pull/141)
